activerecord/lib/active_record/query_logs.rb

```diff
@@ -134,6 +133,7 @@ def tags_formatter=(format) # :nodoc:
         else
           raise ArgumentError, "Formatter is unsupported: #{format}"
         end
+        @tags_formatter = format
```

Comment:
I'd rather this setter be here, after `@formatter`, so that the exception that can be raised above prevents the assignment.